ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/EventsObserver.scala

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.benchtool.submission.EventsObserver._
+import com.daml.ledger.api.benchtool.util.ObserverWithResult
+import com.daml.ledger.api.v1.event.CreatedEvent
+import com.daml.ledger.api.v1.transaction.{TransactionTree, TreeEvent}
+import com.daml.ledger.api.v1.transaction_service.GetTransactionTreesResponse
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.concurrent.Future
+
+object EventsObserver {
```

Comment:
If this code is used only in tests, it should probably go to a new `-test-lib` target.

Reply:
Will be done as part of another PR.